src/Tribe/PUE/Plugin_Info.php

```diff
@@ -88,20 +91,59 @@ public static function from_json( $json ) {
 				return null;
 			}
 
-			//Very, very basic validation.
-			$valid = ( isset( $apiResponse->name ) && ! empty( $apiResponse->name ) && isset( $apiResponse->version ) && ! empty( $apiResponse->version ) ) || ( isset( $apiResponse->api_invalid ) || isset( $apiResponse->no_api ) );
+			// Normalize keys by stripping "plugin_" prefix.
+			$normalizedResponse = [];
+			foreach ( get_object_vars( $apiResponse ) as $key => $value ) {
+				$normalizedKey                        = str_replace( 'plugin_', '', $key );
+				$normalizedResponse[ $normalizedKey ] = $value;
+			}
+
+			// Basic validation after normalization.
+			$valid = ( ! empty( $normalizedResponse['name'] )
+					&& ! empty( $normalizedResponse['version'] ) )
+					|| ( isset( $normalizedResponse['api_invalid'] )
+					|| isset( $normalizedResponse['no_api'] )
+					);
 			if ( ! $valid ) {
 				return null;
 			}
 
-			$info = new Tribe__PUE__Plugin_Info();
-
-			foreach ( get_object_vars( $apiResponse ) as $key => $value ) {
-				$key = str_replace( 'plugin_', '', $key ); // let's strip out the "plugin_" prefix we've added in plugin-updater-classes.
-				$info->$key = $value;
+			// Populate the object.
+			$info = new self();
+			foreach ( $normalizedResponse as $key => $value ) {
+				if ( $info->check_whitelisted_keys( $key ) ) {
+					$info->$key = $value;
+				}
 			}
 
 			return $info;
+			// phpcs:enable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+		}
+
+		/**
+		 * Get the whitelist of valid keys (class properties).
+		 *
+		 * @since TBD
+		 *
+		 * @return array List of valid property names.
+		 */
+		public function get_whitelisted_keys(): array {
+			return array_keys( get_class_vars( __CLASS__ ) );
```

Comment:
Add a filter in case a plugin needs something specific?

Reply:
I do like the idea of adding a filter. However, since this creates the class keys dynamically in  `from_json` it will still result in the error `Deprecated: Creation of dynamic property Tribe__PUE__Plugin_Info::$KEY_ADDED_VIA_FILTER is deprecated in ...`

Reply:
Yeah, I don't think the work-around is worth the extra work.